### PR TITLE
v0.2.3 Flat file data persistence

### LINKED ISSUE #11: HTTPS/WSS SSL support
The server should be modified to support `https://` and `wss://` protocols

### LINKED ISSUE #21: stop command doesn't stop the server
Running stop in the terminal with the server running won't close the server - probably because of netty not being shut down properly.

### LINKED ISSUE #28: Holding down while drawing can also select text
Sometimes when drawing out of bounds text will be selected. Fix by disabled text selection while drawing.

### LINKED ISSUE #31: Remove ConsoleManager
This class is unused and unnecessary. Without this class, a proper shutdown mechanism may be added for netty

### LINKED ISSUE #32: Documents with the same name will not work in the sidebar
SidebarItem.js should use ids instead of names.

### LINKED ISSUE #36: Better config system
Messy program arguments may not be the best way to handle configuring, perhaps switch to a config file or use better documentation?

### PR DESCRIPTION
**Pull request by griffinht**

### CODE REVIEW

File: board-server/src/main/java/net/stzups/board/config/ConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package net.stzups.board.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConfigProvider {
```
Comment (griffinht) [inline board-server/src/main/java/net/stzups/board/config/ConfigProvider.java]: JavaDoc document here please?

File: board-server/src/main/java/net/stzups/board/config/ConfigProviderBuilder.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package net.stzups.board.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (griffinht) [inline board-server/src/main/java/net/stzups/board/config/ConfigProviderBuilder.java:5]: Documentation here too

File: board-server/src/main/java/net/stzups/board/config/configs/ArgumentConfig.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package net.stzups.board.config.configs;
+
+import net.stzups.board.config.StringConfig;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class ArgumentConfig implements StringConfig {
```
Comment (griffinht) [inline board-server/src/main/java/net/stzups/board/config/configs/ArgumentConfig.java:13]: Documentation here

File: board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package net.stzups.board.config.configs;
+
+import net.stzups.board.config.StringConfig;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
```
Comment (griffinht) [inline board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java:9]: Documentation, specify behavior if the given file does not exist

File: board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public class PropertiesConfig implements StringConfig {
+    private Properties properties;
+
+    public PropertiesConfig(String name) throws IOException {
```
Comment (griffinht) [inline board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java]: name parameter should be called path, no?

File: board-server/src/main/java/net/stzups/board/config/ConfigProvider.java (hunk 1 above)
Comment (griffinht) [reply board-server/src/main/java/net/stzups/board/config/ConfigProvider.java]: done

File: board-server/src/main/java/net/stzups/board/config/ConfigProviderBuilder.java (hunk 2 above)
Comment (stzups) [reply board-server/src/main/java/net/stzups/board/config/ConfigProviderBuilder.java:5]: doner

Comment (griffinht) [reply board-server/src/main/java/net/stzups/board/config/ConfigProviderBuilder.java:5]: done

File: board-server/src/main/java/net/stzups/board/config/configs/ArgumentConfig.java (hunk 3 above)
Comment (griffinht) [reply board-server/src/main/java/net/stzups/board/config/configs/ArgumentConfig.java:13]: done

File: board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java (hunk 4 above)
Comment (griffinht) [reply board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java:9]: done

File: board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java (hunk 5 above)
Comment (griffinht) [reply board-server/src/main/java/net/stzups/board/config/configs/PropertiesConfig.java]: done

**Review by griffinht — COMMENTED** [review]
nice

**Pull request merged**